homeassistant/components/tessie/climate.py

```diff
@@ -0,0 +1,129 @@
+"""Climate platform for Tessie integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from tessie_api import (
+    set_climate_keeper_mode,
+    set_temperature,
+    start_climate_preconditioning,
+    stop_climate,
+)
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import TessieDataUpdateCoordinator
+from .entity import TessieEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Tessie Climate platform from a config entry."""
+    coordinators = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(TessieClimateEntity(coordinator) for coordinator in coordinators)
+
+
+class TessieClimateEntity(TessieEntity, ClimateEntity):
+    """Vehicle Location Climate Class."""
+
+    _attr_precision = 0.5
+    _attr_min_temp = 15
+    _attr_max_temp = 28
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_hvac_modes = [HVACMode.HEAT_COOL, HVACMode.OFF]
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_preset_modes: list = ["off", "on", "dog", "camp"]
+
+    def __init__(
+        self,
+        coordinator: TessieDataUpdateCoordinator,
+    ) -> None:
+        """Initialize the Climate entity."""
+        super().__init__(coordinator, "climate")
```

Comment:
climate is redundant here since the unique id is a combination of platform,domain,id but probably ok to leave it since it avoids the complexity of branching

I might call it `primary` instead though in case there will be a climate entity for something else like `driver`/`pax`

Reply:
Unfortuantely the API only allows you to control the drivers side. Primary sounds good.